### PR TITLE
Fixed linking error when Azure SDK is being built as DLL.

### PR DESCRIPTION
**Pull request by antkmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Release History
 
+## 1.3.1 (2021-11-04)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CHANGELOG.md:3]: Let's skip the CL entry change and re-ship 1.3.0?

And if we insist on shipping a 1.3.1, update the date to be 11-05. And maybe do that as a separate PR, so we can get the fix in first and decide separately how/when we wanna ship the fix.

**Review by ahsonkhan — APPROVED** [review]
Other than moving the CL change our, to a separate discussion, looks good.

**Review by vhvb1989 — APPROVED** [review]
